### ISSUE TITLE
potentiality for multiple patch windows in same month?

### ISSUE THREAD

**Issue by xandrellas** [puppetlabs/puppetlabs-patching_as_code#17]
Looking at the is_patchday.rb it appears to follow a monthly-based cycle.  Is there a means to shrink or expand that cycle?  Such as - I'd like this particular patch group to be patched twice a month?

**Comment by kreeuwijk** [comment]
That's interesting for sure. Can you detail what that would look like in your view?
So what you'd like to be able to control, how it should behave and how you'd write it in Hiera.

For example, to (theoretically) patch every 2nd and 4th Thursday of the month:
```
patching_as_code::patch_schedule:
  twice_a_month:
    day_of_week: Thursday
    count_of_week:
        - 2
        - 4
    hours: 10:00 - 11:00
    max_runs: 1
    reboot: ifneeded
```

**Comment by xandrellas** [comment]
Hi Kevin - I also recognize the challenge in attempting to cover as many potential scenarios as possible.
Such as : Patch 2nd Thursday of week at 10:00 - 11:00, patch same group 4th Thursday of week at 06:00 - 08:00
Thankfully that is not the desire of this question/potential-request.

Your theoretical hiera paste above would more than likely be the approach I would consider.

**Comment by kreeuwijk** [comment]
It seems we'd need two new capabilities then:
* Allow multiple values for `count_of_week`
* Allow linking a node to multiple patch schedules

I'll take a look what I can do, this shouldn't be too difficult to implement.

**Comment by kreeuwijk** [comment]
This functionality has now been implemented in version 0.6.0 of the module, available on the Puppet Forge.
